lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -158,10 +102,178 @@ def strip_var_names(cubes):
     return cubes
 
 
+class ConcatenateCubes():
+    """
+    Class adding functionality to iris.concatenate_cubes().
+
+    Accounts for differences in attributes and allows promotion of scalar
+    coordinates to be associated with the dimension over which concatenation
+    is to be performed (eg can promote forecast_period to auxiliary for single
+    time point cube inputs).
+    """
+
+    def __init__(self, master_coord, coords_to_associate=None,
+                 coords_to_slice_over=None):
+        """
+        Initialise parameters
+
+        Args:
+            master_coord (str):
+                Coordinate to concatenate over.
+
+        Kwargs:
+            coords_to_associate (list):
+                List of coordinates to be associated with the master_coord.  If
+                master_coord is "time" this should be "forecast_reference_time"
+                OR "forecast_period", NOT both.
+            coords_to_slice_over (list):
+                Dimension coordinates to slice over before concatenation.
+        """
+        self.master_coord = master_coord
+        self.coords_to_associate = coords_to_associate
+        self.coords_to_slice_over = coords_to_slice_over
+
+        if self.coords_to_slice_over is None:
+            self.coords_to_slice_over = ["realization", "time"]
+        if self.coords_to_associate is None and self.master_coord == "time":
+            self.coords_to_associate = ["forecast_period"]
+
+        self.silent_attributes = ["history", "title", "mosg__grid_version"]
```

Comment:
Possibly.  My aim here was to mimic what was being done by the old functions, where these attributes are hard-coded into the `_equalise_cube_attributes` function.  What I didn't want to set up at this stage was a default argument including an `mosg__` attribute, as that doesn't seem to me to be consistent with some of the metadata work we're doing - and actually the metadata work might end up changing the list of "silent" attributes anyway.  So I think in the longer term, yes this might be sensible, but I'm not sure we're in a good position to do that just yet.

What do you think?

Reply:
Alright. I'm happy for this to be defined here for the moment. Is it possible to record this somewhere as some metadata work that we might want to revisit? I think it would be worth adding a comment here to explain why you're defining this variable here. Something similar to your docstring for the `equalise_cube_attributes` function i.e. `List of attributes to remove silently if unmatched` would be fine.

Reply:
I've added a note to epic IMPRO-1107 ("correct global attributes") that the silent attributes should be reviewed here.

Reply:
Thanks @cgsandford